adr/2024-09-states.md

```diff
@@ -0,0 +1,76 @@
+
+### Storage of state specific copy
+
+Copy used in iv-verify shouuld be configurable by state. Additionally all copy should be able to be translated by the iv-verify i18n system. The decision of how to store the state specific copy is to either store the text in the same file with different keys or in separate files with the same key. In both options a change needs to be made to the `t()` function to look up state specific copy if it exists.
+
+#### Separate file - different keys
+
+In this option the text for state specific copy would be stored together. The `/app/i18n/locales/en/translation.json` file would have keys like: `add_expense_amount_field` and then state specific overrides like: `add_expense_amount_field__or` where "or" is the state key for Oregon. The `t()` function would keep track of which state configuration is being used and look up the state specific key if that doesn't exist it would fallback to the regular key.
+
+#### Separate files - same key
+
+In this option the text for the state specific copy is stored in spearate files such as `/app/i18n/locales/en/or.json`. This file would include keys such as `add_expense_amount_field` that overwrite the values in the `/app/i18n/locales/en/translation.json` file. The `t()` function would first look to the state configuration file before looking to the fallback file. This functionality does not exist in the i18n next infrastructure and would need to be developed.
+
+### Routing between configurations
+
+Once an end user arrives on iv-verify they need to be able to start an application for a particular state. Which state a user is using can be configured in either the URL or in a Cookie.
+
+#### URL
+
+In this option the URL for the ledger add screen goes from https://verify-prototype.app.cloud.gov/ledger/income/add to https://verify-prototype.app.cloud.gov/or/ledger/income/add where "or" is the state configuration key (OR for Oregon). This option makes it very clear which flow you are in and can make testing easier. The downside is that it requires a bit of work to modify the URLRouter to generate the URLs correctly to navigate between screens.
+
+#### Cookie
+
+In this option the URL for the ledger add screen states the same but when the user initiates the flow a cookie is set to indicate which flow the user is in. Both the client and server will have access to this cookie to navigate the user down the flow. The downside to this solution is that is not visibly clear which flow a user is unless the UI is modified to say so.
+
+## Decision
+
+The recommendations are as follows:
+
+### Storage
+
+Configuration should be stored in files for iv-verify. This requires the least upfront work and provides version control for configuration changes. iv-verify is not at the point where making configuration changes without a deploy is necessary.
```

Comment:
i concur on this point for the early stages. eventually we may want to move to something else, but for an MVP, this is where i landed as well

Reply:
Here is [our configuration file for reference](https://github.com/DSACMS/iv-cbv-payroll/blob/main/app/config/site-config.yml) if it's helpful. We have a mix of checking in configuration as YAML and using environment variable interpolation.

Reply:
thank you thank you!